server/app/controllers/admin/AdminImportController.java

```diff
@@ -126,163 +126,165 @@ public Result hxImportProgram(Http.Request request) {
-            program.adminName(),
-            program.localizedName().getDefault(),
-            program.localizedShortDescription().getDefault(),
-            program.externalLink(),
-            program.displayMode().getValue(),
-            notificationPreferences,
-            ImmutableList.of(), // categories are not migrated
-            ImmutableList.of() // associated TI groups are not migrated
-            );
-    if (!programErrors.isEmpty()) {
-      // We want to reference "admin name" instead of "URL" in errors, because there is no URL field
-      // in program migration. The existing error strings were created for the program create/edit
-      // UI which has a URL field that is generated from the admin name.
-      String errorString = joinErrors(programErrors).replace("URL", "admin name");
-      return ok(
-          adminImportViewPartial
-              .renderError("One or more program errors occured:", errorString)
-              .render());
-    }
+      // Check that all block definition ids are positive numbers
+      for (BlockDefinition blockDefintion : program.blockDefinitions()) {
+        long blockId = blockDefintion.id();
+        if (blockId < 1) {
+          return ok(
+              adminImportViewPartial
+                  .renderError(
+                      "Block definition ids must be greater than 0.",
+                      "Please check your block definition ids and try again.")
+                  .render());
+        }
+      }
 
-    boolean withDuplicates = !settingsManifest.getNoDuplicateQuestionsForMigrationEnabled(request);
-
-    // When we are importing without duplicate questions, we expect all drafts to be published
-    // before the import process begins.
-    Optional<VersionModel> draftVersion = versionRepository.getDraftVersion();
-    if (!withDuplicates
-        && draftVersion.isPresent()
-        // If there are either questions or programs in the draft, we should show this error
-        && (versionRepository.getProgramCountForVersion(draftVersion.get())
-                + versionRepository.getQuestionCountForVersion(draftVersion.get())
-            > 0)) {
-      return ok(
-          adminImportViewPartial
-              .renderError(
-                  "There are draft programs and questions in our system.",
-                  "Please publish all drafts and try again.")
-              .render());
-    }
+      // Check for other validation errors like invalid program admin names
+      ImmutableList<String> notificationPreferences =
+          program.notificationPreferences().stream()
+              .map(preference -> preference.getValue())
+              .collect(ImmutableList.toImmutableList());
+
+      ImmutableSet<CiviFormError> programErrors =
+          programService.validateProgramDataForCreate(
+              program.adminName(),
+              program.localizedName().getDefault(),
+              program.localizedShortDescription().getDefault(),
```

Comment:
This `getDefault` is what's throwing a server error when the default translation doesn't exist. There's already a test for catching server errors [here](https://github.com/civiform/civiform/blob/551b2faa5976c92174cc05d76b8e56e42fef0112/server/test/controllers/admin/AdminImportControllerTest.java#L298) and I opted not to add a separate test for this case because it would involve adding another giant block of json. But lmk if you think that's important to have and I can add it.